ui-modules/blueprint-composer/app/components/providers/blueprint-service.provider.js

```diff
@@ -43,6 +43,9 @@ export const COMMON_HINTS = {
     }]
 };
 
+// needed to support error messages for extra constraints
+export const EXTRA_CONSTRAINTS = ['equal', 'greater_than', 'greater_or_equal', 'less_than ', 'less_or_equal ', 'in_range ', 'valid_values', 'length ', 'min_length ', 'max_length ', 'pattern'];
```

Comment:
Please add references to implementation of these extra constraints, either in the Brooklyn core or its dependencies.

Reply:
Also the trailing whitespace in some is odd.  Do we trim so it does not matter?